contracts/walletCache.sol

```diff
@@ -0,0 +1,81 @@
+/**
+ *  The Consumer Contract Wallet Deployer
+ *  Copyright (C) 2019 The Contract Wallet Company Limited
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+pragma solidity ^0.5.10;
+
+import "./wallet.sol";
+import "./internals/ensResolvable.sol";
+
+
+/// @title IWalletCache interface describes a method for poping an already cached wallet
+interface IWalletCache {
+    function walletCachePop() external returns(Wallet);
+}
+
+//// @title Wallet cache with wallet pre-caching functionality.
+contract WalletCache is ENSResolvable{
+
+    event CachedWallet(Wallet _wallet);
+
+    /*****   Constants   *****/
+    bytes32 public constant controllerNode = 0x7f2ce995617d2816b426c5c8698c5ec2952f7a34bb10f38326f74933d5893697;
```

Comment:
this is same thing as in the current wallet deployer - should be provided in the constructor, otherwise we won't be able to test these contracts on different ENS names

Reply:
This has been changed, it looks good to me now @draganm ... am going to merge it, but let me know your thoughts nonetheless.